### ISSUE TITLE
[Portainer][Error Code 400] Unable to deploy with Portainer

### ISSUE THREAD

**Issue by Pheggas** [r7wx/easy-gate#12]
Hello. I'm pretty sure the fault is on my side somewhere but i'm unable to deploy Easy Gate via Portainer. Don't get me wrong, i already spun up more than 20 containers but i tried to deploy this one for 3 times now and got error with code 400 after each try.

Any clues?

**Comment by r7wx** [comment]
Hello, could you provide the following info?
- Easy Gate version.
- Easy Gate configuration file.

If the 400 error comes from Easy Gate itself there is only one reason for this:

```go
reqIP, _, err := net.SplitHostPort(req.RemoteAddr)
if cfg.BehindProxy {
    reqIP = req.Header.Get("X-Forwarded-For")
    if reqIP == "" {
	http.Error(w, "", http.StatusBadRequest)
	return
   }
}
```

So probably you have configured Easy Gate to run behind an already configured reverse proxy, but you are not providing the "X-Forwarded-For" header correctly. If you are using nginx you can find an example configuration in examples folder or in the main Readme:

```nginx
[...]
location / {
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_pass http://easy-gate:8080;
}
[...]
```

**Comment by Pheggas** [comment]
Version of EasyGate is latest (1.1.1) and configuration file wasn't even created. But I probably caught the issue already. On host part, i defined only the folder, not the config file so that could be the issue. I will test it when I get home.

In any case I will edit this post.

**Comment by Pheggas** [comment]
I just tried what i thought would be the fix but unfortunately, still the same error code 400. And it's not shown by EasyGate but by portainer.

**Comment by Pheggas** [comment]
Another update, i tried to run it as docker compose file in your README. I found out portainer has issue with specifying exact file. When i defined only folder, not that configuration file, the container stared. But Web UI wasn't avalible. I headed into logs and saw `open /etc/easy-gate/easy-gate.json: no such file or directory`.

This way i wanted to `docker exec` into that container and add the provided config file provided in this repo. I wasn't able to do so because container keep restarting (till the death i think). Any ideas how could i import config file inside container while it's restarting or it's stopped?
